stream.go

```diff
@@ -0,0 +1,92 @@
+package esja
+
+import (
+	"errors"
+	"fmt"
+)
+
+// Stream represents a queue of events and basic stream properties.
+// Zero-value is a valid state, ready to use.
```

Comment:
perhaps not anymore, if there's `id` inside now :(

Reply:
and `streamType`

Reply:
You are right! I will fix that.

Reply:
https://github.com/ThreeDotsLabs/esja/pull/20/commits/4072c2da705e2a3c2aab7988b9c33c871d990eaf